### PR TITLE
Add test for content app range requests

### PR DESCRIPTION
**Pull request by dralley**
[noissue]

### CODE REVIEW

**Comment by dralley** [comment]
So this definitely reproduces on master.  I'm not exactly sure why it didn't originally.

File: pulpcore/tests/functional/api/using_plugin/test_distributions.py [hunk 1]
```diff
@@ -276,91 +279,127 @@ class ContentServePublicationDistributionTestCase(unittest.TestCase):
...
-        distro = self.dis_api.read(monitor_task(distro_response.task).created_resources[0])
-        if cleanup:
-            cleanup(self.repo_api.delete, repo.pulp_href)
-            cleanup(self.remote_api.delete, remote.pulp_href)
-            cleanup(self.dis_api.delete, distro.pulp_href)
-            if publish:
-                cleanup(self.pub_api.delete, pub.pulp_href)
-        if publish:
-            return repo, remote, pub, distro
-        return repo, remote, distro
```
Comment (dralley) [inline pulpcore/tests/functional/api/using_plugin/test_distributions.py:361]: Apologies for rewriting the entire test, I had actually started on the 3.11 branch and then moved it to master.  But this abstraction doesn't work all that well if you need to tweak individual parameters anyway, unfortunately.

**Review by bmbouter — APPROVED** [review]
This looks good. Thank you!

**Review by goosemania — APPROVED** [review]
Looks good!
I take it as a prep to fix https://pulp.plan.io/issues/8865

**Comment by bmbouter** [comment]
> Looks good!
> I take it as a prep to fix https://pulp.plan.io/issues/8865

Yeah exactly. I'm hoping to look at fixing that (with this test) unless someone else beats me to it.

**Pull request merged**